### PR TITLE
Add custom template configuration and documentation to tech-guide

### PR DESCRIPTION
**Pull request by cosletmihai**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -30,4 +30,20 @@ start-services: set-linekdpipes-etl-configurations
 
 stop-services:
 	@ echo "$(BUILD_PRINT)Stopping the Docker compose services"
-	@ docker-compose --file docker/docker-compose.yml --env-file docker/.env down
\ No newline at end of file
+	@ docker-compose --file docker/docker-compose.yml --env-file docker/.env down
+
+#-----------------------------------------------------------------------------
+# Template commands
```
Comment (costezki) [inline Makefile:36]: we need the other two commands as well, don't we?

File: Makefile [hunk 2]
```diff
@@ -30,4 +30,20 @@ start-services: set-linekdpipes-etl-configurations
...
+#-----------------------------------------------------------------------------
+build-template-volumes:
+	@ docker volume create rdf-differ-template
+
+differ-set-report-template:
+	@ [ "$(location)" ] || ( echo ">> template 'location' is not set"; exit 1 )
```
Comment (costezki) [inline Makefile:42]: nice hack

File: docker/.env [hunk 3]
```diff
@@ -4,6 +4,7 @@ RDF_DIFFER_UI_PORT=8030
 RDF_DIFFER_API_PORT=4030
 RDF_DIFFER_API_LOCATION=http://rdf-differ-api
 RDF_DIFFER_GUNICORN_TIMEOUT=1200
+RDF_DIFFER_TEMPLATE_LOCATION=/usr/src/template
```
Comment (costezki) [inline docker/.env:7]: Important to mention in the documentation that this si the **internal** path

File: docker/docker-compose.yml [hunk 4]
```diff
@@ -248,6 +250,8 @@ volumes:
   linkedpipes-configuration:
     name: linkedpipes-configuration
     external: true
+  rdf-differ-template:
+    external: true
```
Comment (costezki) [inline docker/docker-compose.yml:254]: yup

File: docs/tech-manual/content/configuration.tex [hunk 5]
```diff
@@ -58,6 +58,49 @@ \section{Configuration}
...
+		\end{lstlisting}
+
+		\textbf{NOTE}: Make sure that the location specified ends with a trailing slash \texttt{/}, otherwise the command will not work propery and the templates will not be copied to the docker volume.
+
+		Example:
+		\begin{lstlisting}[language=bash]
+make location=~/template/location/ differ-set-report-template
+		\end{lstlisting}
+		 
+		\subsubsection{Use the custom template}
```
Comment (costezki) [inline docs/tech-manual/content/configuration.tex:76]: same title as above

File: docs/tech-manual/content/configuration.tex [hunk 6]
```diff
@@ -58,6 +58,49 @@ \section{Configuration}
...
+		An externally defined volume \texttt{rdf-differ-template} which will contain the externally defined template (aka: the custom template), which in turn is coupled with the \texttt{rdf-differ-api} docker container to use when generating the reports.
+		\begin{lstlisting}[]
+volumes:
+- rdf-differ-template:${RDF_DIFFER_TEMPLATE_LOCATION}
+		\end{lstlisting}
+		The lines above map the custom template that has been copied to the docker volume with the internal location of the container which has been defined in the \texttt{.env} file.
+		
+		To configure your own template you can copy the default report template and adjust it to your needs. Read more about the required structure of the template on the \href{https://eds4jinja2.readthedocs.io/en/latest/#target-directory-layout}{\underline{eds4jinja2}} documentation page.
+ 
+		\subsubsection{Use the custom template}
```
Comment (costezki) [inline docs/tech-manual/content/configuration.tex]: same title as above

**Review by costezki — APPROVED** [review]
well done, 

- we still need to add the rest of the make targets
- and for the documentation, I will polish the text after it is merged.

File: Makefile (hunk 1 above)
Comment (cosletmihai) [reply Makefile:36]: yes, they will be added. I implemented everything for `rdf-differ`, and then I'll replicate the same thing for the other 2

**Review by costezki — APPROVED** [review]

**Review by costezki — APPROVED** [review]

**Pull request merged**